juju/juju
# Add initial jujud operator

## Description of change

Add a skeleton jujud operator. The operator starts essentially a no-op worker which can be fleshed out going forward. Some small common bits from uniter were extracted out, namely the symlink creation.

The CAAS provisioner facade and state model gets support for setting a password on the application entity. This will allow the operator via the associated facade (not yet written) to make api calls back to the controller.

The k8s client is improved to delete and re-create config map and pod so allow for changed config to be applied as needed. The pod delete gets a retry loop to allow for the terminating phase to complete.

## QA steps

Run up a CAAS model and deploy a charm.
Run upgrade-juju and check that the operator pod is deleted and restarted.
